### DISCUSSION TITLE
트러블 슈팅 : -4094 에러 해결 과정

### DISCUSSION THREAD

**Discussion by slsl2** [nbc-final-a10/TripBuddy#96] [자유게시판]
좋은 에러 해결 과정이라고 생각해서 블로그에 쓴 글 그대로 공유합니다!
원문: https://velog.io/@minn__ij/%EB%82%B4%EB%B0%B0%EC%BA%A0TIL-15%EC%A3%BC%EC%B0%A8-%ED%99%94%EC%9A%94%EC%9D%BC

# 트러블 슈팅

## 🚨 문제

> **_-4094 에러 (Internal Server Error)_**
![](https://velog.velcdn.com/images/minn__ij/post/4ab3acdc-216c-41c9-bc8e-e08cd6deea3c/image.png)

Next.js 를 사용하기 시작하고 나를 계속 괴롭힌 에러.

간단한 CSS 수정을 하든 뭘 하든 뭘 안 하든 짧게는 몇십초 길게는 몇분 내에 꼭 저 에러가 뜨면서 랜더링이 멈추고, yarn dev를 다시 실행해야 했다.

### **1. .next, node_modules 폴더 삭제 후 yarn 패키지 재설치**

→ 해결 실패

### **2. 구글링**

→ 삭제 후 재설치 외에 '운영체제 초기화' 라는 해결책을 발견…

→ 다른 방법 찾아보기로 결정

### **3. GPT에게 도움 요청**

→ 권한 문제일 수도 있다는 답변!

> _파일 시스템 권한 확인 권한 문제일 수도 있으므로, 프로젝트 폴더와 하위 파일/폴더에 대한 읽기/쓰기 권한을 확인합니다._
```
# Windows에서는 관리자 권한으로 실행해야 할 수도 있습니다
icacls "프로젝트경로" /grant Everyone:(OI)(CI)F /T
```
_위 명령어는 프로젝트 폴더와 하위 파일/폴더에 대해 모든 사용자에게 읽기/쓰기 권한을 부여합니다._

→ 해결

→ 된 줄 알았지만… 몇 시간은 괜찮다가 작업을 계속하다 보니 다시 똑같은 에러 발생.

### **1. 에러가 발생하는 시점이 언제인지 터미널을 다시 살펴보았다.**

VSCode에서 저장됨 > 자동으로 반영 > 서버에 띄워줌 이런 식으로 작동되는데

이게 Compiling이 잘 되다가.. 잘 되다가… 갑자기 뜬금없이 에러가 생긴다.

![](https://velog.velcdn.com/images/minn__ij/post/f0e036ea-2b41-4584-85cb-87fa9755ba87/image.png)

### 2. 원인 추측

> **_지금 글자 하나 하나 바뀔 때마다 자동으로 저장이 되는데, 렌더링은 그 속도를 못 따라가서 생기는 문제가 아닐까..???_**

### **3. VS Code 설정을 변경해보자..**

원래 코드 변경이 있을 때마다 자동 저장 되도록 설정을 해 놨었음.
이게 문제 될 거라고는 생각해본 적이 없는데, 문득 그럴 수도 있겠다는 생각이 들음.
Auto Save: afterDelay로 되어 있던 걸 off로 변경해봄.

![](https://velog.velcdn.com/images/minn__ij/post/201b1088-821a-4819-ad4e-3ce2fa09920c/image.png)

### 4. 해결.

이렇게 간단한 문제였다니… 상상도 못 했다.
그래도 혼자의 힘으로 간단하지만 **엄청나게 중요한** 문제를 해결해서 뿌듯한 경험이 됐다.
컴퓨터 속도 등 환경에 따라 다를 수도 있지만, 이런 방법도 있다는 점을 기록해 놓고 싶다.

## 👍🏻 결론
> **VS Code 설정에 자동저장이 켜져 있으면 꺼보세요!!**

**Comment by Captain-Kim** [comment]
이런 에러는 본 적이 없지만 접근 방법이 너무 인상적입니다 최곱니다

**Comment by eunohhh** [comment]
맞아요~~! 매우 꿀팁입니다 감사합니다!
